cluster/terraform_kubernetes/kube_state_metrics.tf

```diff
@@ -0,0 +1,219 @@
+  }
+  rule {
+    api_groups = ["coordination.k8s.io"]
+    resources  = ["leases"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["autoscaling"]
+    resources  = ["horizontalpodautoscalers"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["policy"]
+    resources  = ["poddisruptionbudgets"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["batch"]
+    resources  = ["cronjobs"]
+    verbs      = ["get", "list", "watch"]
+  }
+  rule {
+    api_groups = ["", "apps", "batch", "networking.k8s.io", "policy", "autoscaling", "certificates.k8s.io", "coordination.k8s.io", "storage.k8s.io"]
+    resources  = ["pods", "replicasets", "cronjobs", "ingresses", "poddisruptionbudgets", "networkpolicies", "storageclasses", "certificatesigningrequests", "leases", "horizontalpodautoscalers", "configmaps", "secrets", "nodes", "services", "resourcequotas", "replicationcontrollers", "limitranges", "persistentvolumeclaims", "persistentvolumes", "namespaces", "endpoints", "deployments", "statefulsets", "daemonsets", "volumeattachments", "mutatingwebhookconfigurations", "validatingwebhookconfigurations", "jobs"]
+    verbs      = ["get", "list", "watch"]
+  }
+
+}
+resource "kubernetes_cluster_role_binding" "kube_state_metrics" {
+  metadata {
+    name = "kube-state-metrics"
+    labels = {
+      "app.kubernetes.io/component" = "exporter"
+      "app.kubernetes.io/name"      = "kube-state-metrics"
+      "app.kubernetes.io/version"   = var.kube_state_metrics_version
+    }
+  }
+
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.kube_state_metrics.metadata[0].name
+  }
+
+  subject {
+    kind      = "ServiceAccount"
+    name      = "kube-state-metrics"
+    namespace = "monitoring"
+  }
+}
+
+resource "kubernetes_deployment" "kube_state_metrics" {
+  metadata {
+    name      = "kube-state-metrics"
+    namespace = "monitoring"
+    labels = {
+      "app.kubernetes.io/component" = "exporter"
+      "app.kubernetes.io/name"      = "kube-state-metrics"
+      "app.kubernetes.io/version"   = var.kube_state_metrics_version
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        "app.kubernetes.io/name" = "kube-state-metrics"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          "app.kubernetes.io/component" = "exporter"
+          "app.kubernetes.io/name"      = "kube-state-metrics"
+          "app.kubernetes.io/version"   = var.kube_state_metrics_version
+        }
+      }
+
+      spec {
+        automount_service_account_token = true
+        service_account_name            = kubernetes_service_account.kube_state_metrics.metadata[0].name
+
+        container {
+          name  = "kube-state-metrics"
+          image = "k8s.gcr.io/kube-state-metrics/kube-state-metrics:v${var.kube_state_metrics_version}"
```

Comment:
it looks like k8s.gcr.io is the old repository and they moved to registry.k8s.io

Reply:
done